### DISCUSSION TITLE
Function running on different cycle

### DISCUSSION THREAD

**Discussion by Allanadn** [bitcraze/discussions#982] [Q&A]
Dear Crazyflie community,

I am new to the crazyflie UAV and I'd like to ask a question. When using "commander.send_position_setpoint" to send commands from a motion planner for the crazyflie to follow, the crazyflie behaves very strangely. It takes off and after going back and forth for a couple of times it drops down on the ground. I have also used to the go.to method, which makes the crazyflie follow very slowly the path proposed (even if I increase the default speed). Any ideas on why this is happenning and any better methods to be used? (I am currently using the flowdeck v2 to estimate the CF position). 

Thanks in advance

**Comment by krichardsson** [comment]
Have you tried to fly using the buttons in the "Command based flight control" box in the "Flight control" tab in the client, or run one of the example scripts in the python lib repository, for instance [motion_commander_demo.py](https://github.com/bitcraze/crazyflie-lib-python/blob/ea0154e68ad4ae6a6deda3d4ffb5f1011efed829/examples/autonomy/motion_commander_demo.py), to make sure positioning is OK?

If your positioning is OK, I think your problem might be that you send too few setpoints. There is a safety mechanism in the Crazfylie that expects setpoints continuously to make sure there is still a connection to the ground station. If you send 4-5 setpoints/s you should be good, just resend the same one over and over if you don't want to change the values. 
After landing you also need to call `cf.commander.send_notify_setpoint_stop()` as described in the [documentation](https://github.com/bitcraze/crazyflie-firmware/blob/master/docs/functional-areas/sensor-to-control/commanders_setpoints.md#setpoint-priority) to be able to take off again

**Reply by Allanadn** [reply]
Thank you for your answer Kristoffer and sorry for the long time to get back to you. I have double checked the positioning and they seem to be ok. I have adjusted the number of points sent per second and it indeed helped a lot in the reduction of the instances where the UAV just dropped to the ground. When it came to send velocity waypoints to the commander module (bypassing the High Level commander), the UAV never fell unexpectedly to the ground again. Nonetheless, when sending position waypoints to the commander module (bypassing the High Level commander), the previously seen behavior persists (very slow moviment of the UAV until it drops unexpectedly to the ground). Is there a different rate expected by the crazyflie when sending velocity or position waypoint commands?

Thanks again for the help

**Reply by krichardsson** [reply]
> Is there a different rate expected by the crazyflie when sending velocity or position waypoint commands?

No, it is the same timeout regardless of setpoint type

**Reply by Allanadn** [reply]
I see. Well, not sure what could cause this then, since the internal timeout on my algorithm (regardless of setpoint type) is now set to 0.15s. I'll keep investigating. Thank you!
